modules/OFConnectionManager2/module/src/ofconnectionmanager.c

```diff
@@ -400,9 +403,66 @@ verify_cert(SSL_CTX *ctx, const char *cert_filename)
 }
 
 
+/* true if str ends with searchstr  */
+static bool
+ends_with(char *str, char *searchstr)
+{
+    char *substr = strstr(str, searchstr);
+    return substr && (*(substr + strlen(searchstr)) == '\0');
```

Comment:
This could fail in an edge case where the needle occurs more than once in the string. Suggest strcmp at `str+strlen(str)-strlen(searchstr)` (assuming str is at least as long as searchstr).